pkg/controller/csr/csr_controller.go

```diff
@@ -0,0 +1,223 @@
+package csr
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math"
+	"math/big"
+	"time"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/controller/certificatemanager"
+	"github.com/tigera/operator/pkg/controller/options"
+	"github.com/tigera/operator/pkg/controller/utils"
+	"github.com/tigera/operator/pkg/tls"
+	"github.com/tigera/operator/pkg/tls/certificatemanagement"
+	certificatesv1 "k8s.io/api/certificates/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/client-go/kubernetes"
+	typedcertificatesv1 "k8s.io/client-go/kubernetes/typed/certificates/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var log = logf.Log.WithName("controller_csr")
+var controllerName = "csr-controller"
+
+// LabelName label that we set on our CSRs, so it helps us narrow down our list somewhat.
+const LabelName = "k8s-app"
```

Comment:
Maybe this should be something less generic? like `operator.tigera.io/csr`? 

Does this need to be public?

Reply:
I think we could probably do it, or perhaps just remove the whole label name check altoghether.

If we wanna change it, we have to also make a change in https://github.com/tigera/key-cert-provisioner/blob/3abd79340d97354aa6b24789b302664795c324e0/pkg/k8s/certificate.go#L104

Reply:
Added to a PR for key-cert-provisioner: https://github.com/tigera/key-cert-provisioner/pull/89